File: Yet another query problem.cpp
class Solution {
public:
    vector<int> solveQueries(int N, int num, vector<int> &A, vector<vector<int>> &Q) {
        vector<int> ans;
        vector<vector<int>> pre(N+1,vector<int>(N+1,0));
        for(int i=0;i<N;i++){
            int cnt=0;
            for(int j=i;j<N;j++){
                if(A[i]==A[j]){
                    cnt++;
                }
            }
            pre[i][cnt]++;
        }
        for(int i=0;i<N+1;i++){
            for(int j=1;j<N;j++){
                pre[j][i]+=pre[j-1][i];
            }
        }
        for(int i=0;i<num;i++){
            int L=Q[i][0];
            int R=Q[i][1];
            int K=Q[i][2];
            ans.push_back((L==0?pre[R][K]:pre[R][K]-pre[L-1][K]));
        }
        return ans;
    }
};

File: Yet another query problem.py
from typing import List

class Solution:
    def solveQueries(self, N : int, num : int, A : List[int], Q : List[List[int]]) -> List[int]:
        ans=[]
        pre=[[0]*(N+1) for _ in range(N+1)]
        for i in range(N):
            cnt=0
            for j in range(i,N):
                if A[i]==A[j]:
                    cnt+=1
            pre[i][cnt]+=1
        for i in range(N):
            for j in range(1,N):
                pre[j][i]+=pre[j-1][i]
        for i in range(num):
            L=Q[i][0]
            R=Q[i][1]
            K=Q[i][2]
            if L==0:
                ans.append(pre[R][K])
            else :
                ans.append(pre[R][K]-pre[L-1][K])
        return ans
